### PR TITLE
Add payment metadata functionality to sdk-cli

### PR DESCRIPTION
**Pull request by hydra-yse**
_Note:_ Had to make some changes to check whether or not the payment exists, due to the fact that there is no foreign key relation on the `payments_metadata` table

### CODE REVIEW

File: tools/sdk-cli/src/commands.rs [hunk 1]
```diff
@@ -144,6 +144,16 @@ pub(crate) enum Commands {
         /// Optional offset in payments
         #[clap(short = 'o', long = "offset")]
         offset: Option<u32>,
+
+        /// Optional metadata filter, in the form of json_path:json_value,...
+        #[clap(short = 'm', long = "metadata", num_args = 1.., value_delimiter = ',')]
```
Comment (dangeross) [inline tools/sdk-cli/src/commands.rs:149]: Can't the value_delimiter `,` and the split token `:`  be a valid components of json_value? 

E.g. `supportsNested:{"value":[1,2]}` or `supportsNested.value:[1,2]`

Comment (hydra-yse) [reply tools/sdk-cli/src/commands.rs:149]: True, that'd be the case for objects/arrays. Perhaps a delimiter isn't necessary, and we can use `-m` repeatedly. Concerting the `:`, I'll just join the slices into a single string. That should avoid the issue.

File: tools/sdk-cli/src/command_handlers.rs [hunk 2]
```diff
@@ -215,11 +215,30 @@ pub(crate) async fn handle_command(
...
+                Some(raw_filters) => {
+                    let mut filters = vec![];
+
+                    for filter in raw_filters.iter() {
+                        let i = filter.find(':').context("Invalid metadata filter")?;
+
+                        filters.push(MetadataFilter {
+                            json_path: filter[0..i].to_string(),
+                            json_value: filter[i + 1..].to_string(),
+                        });
```
Comment (hydra-yse) [inline tools/sdk-cli/src/command_handlers.rs:230]: Perhaps the more proper solution @dangeross. Let me know what you think

Comment (dangeross) [reply tools/sdk-cli/src/command_handlers.rs:230]: Or maybe `filter.split_once(':')`?

Comment (hydra-yse) [reply tools/sdk-cli/src/command_handlers.rs:230]: Guess that's just more Rust magic to know of. Fixed.

**Review by dangeross — APPROVED** [review]
LGTM!

**Pull request merged**